source/containers.tex

```diff
@@ -25257,38 +25262,163 @@
   \end{itemize}
-\item let \tcode{rank} be the number of elements in \tcode{slices};
-\item let $s_k$ be the $k^\text{th}$ element of \tcode{slices};
-\item let $S_k$ be the type of $s_k$; and
-\item let \tcode{\placeholder{map-rank}} be an \tcode{array<size_t, rank>} such that
-for each $k$ in the range \range{0}{rank},
-\tcode{\placeholder{map-rank}[$k$]} equals:
+\end{itemize}
+
+\pnum
+Given a signed or unsigned integer type \tcode{IndexType},
+a type $S$ is a
+\defnx{canonical \tcode{submdspan} index type for \tcode{IndexType}}{\tcode{submdspan}!canonical index type}
+if $S$ is either \tcode{IndexType} or \tcode{constant_wrapper<V>}
+for some value \tcode{v} of type \tcode{IndexType},
+such that \tcode{v} is greater than or equal to zero.
+
+\pnum
+Given a signed or unsigned integer type \tcode{IndexType},
+a type $S$ is a
+\defnx{canonical \tcode{submdspan} slice type for \tcode{IndexType}}{\tcode{submdspan}!canonical slice type}
+if exactly one of the following is \tcode{true}:
+\begin{itemize}
+\item
+  $S$ is \tcode{full_extent_t};
+\item
+  $S$ is a canonical \tcode{submdspan} index type for \tcode{IndexType}; or
+\item
+  $S$ is a specialization of \tcode{strided_slice}
+  where all of the following hold:
   \begin{itemize}
   \item
-  \tcode{dynamic_extent}
-  if $S_k$ models \tcode{\libconcept{convertible_to}<index_type>},
+    \tcode{S::offset_type}, \tcode{S::extent_type}, and \tcode{S::stride_type}
+    are all canonical \tcode{submdspan} index types for \tcode{IndexType}; and
   \item
-  otherwise,
-  the number of types $S_j$ with $j < k$ that
-  do not model \tcode{\libconcept{convertible_to}<index_type>}.
+    if \tcode{$S$::stride_type} and \tcode{$S$::extent_type}
+    are both specializations of \tcode{constant_wrapper},
+    then \tcode{$S$::stride_type::value} is greater than zero.
+  \end{itemize}
+\end{itemize}
+
+\pnum
+A type $S$ is a \defnadj{collapsing}{slice type} if
+it is neither \tcode{full_extent_t} nor
+a specialization of \tcode{strided_slice}.
+\begin{note}
+Each collapsing slice type in \tcode{submdspan_mapping}'s parameter pack
+of slice specifier types
+reduces the rank of the result of \tcode{submdspan_mapping} by one.
+\end{note}
+
+\pnum
+A type \tcode{S} is a \defnadj{unit-stride}{slice type} if
+\begin{itemize}
+\item
+  \tcode{S} is a specialization of \tcode{strided_slice}
+  where \tcode{S::stride_type} is a specialization of \tcode{constant_wrapper} and
+  \tcode{S::stride_type::value} is equal to 1, or
+\item
+  \tcode{S} denotes \tcode{full_extent_t}.
+\end{itemize}
+
+\pnum
+Given an object \tcode{e} of type \tcode{E} that is a specialization of \tcode{extents}, and
+an object \tcode{s} of type \tcode{S}
+that is a canonical \tcode{submdspan} slice type for \tcode{E::index_type},
+the \defnx{\tcode{submdspan} slice range of \tcode{s} for the $k^\text{th}$ extent of \tcode{E}}{\tcode{submdspan}!slice range}
+of \tcode{e} is:
```

Comment:
You've got "extent of `E` as part of the defined term, then "of `e`" following it.

It's supposed to be "extent of `e`" in the defined term (lowercase), and nothing following it.